src/p2p/Archivers.ts

```diff
@@ -425,6 +433,10 @@ export function addLeaveRequest(leaveRequest: P2P.ArchiversTypes.Request, tracke
       reason: 'This archiver leave request timestamp exceeds acceptable timestamp range',
     }
   }
+  if (!crypto.verify(leaveRequest, leaveRequest.nodeInfo.publicKey)) {
```

Comment:
Please compare the sig in the leave request to the known public key for the archiver that is requesting to leave. We don't want to verify the sig against an untrusted key that is included in the request. Simply checking that the pub key is present in the active archivers list is not enough, we need to check that the key we know for that archiver is the one that signed the leaveRequest.